isaaclab_arena/tasks/lift_object_task.py

```diff
@@ -0,0 +1,238 @@
+
+    def make_termination_cfg(self):
+        object_dropped = TerminationTermCfg(
+            func=mdp_isaac_lab.root_height_below_minimum,
+            params={
+                "minimum_height": self.background_scene.object_min_z,
+                "asset_cfg": SceneEntityCfg(self.lift_object.name),
+            },
+        )
+        return LiftObjectTerminationsCfg(object_dropped=object_dropped)
+
+    def get_events_cfg(self):
+        return self.events_cfg
+
+    def get_mimic_env_cfg(self, embodiment_name: str):
+        raise NotImplementedError("Function not implemented yet.")
+
+    def get_metrics(self) -> list[MetricBase]:
+        return []
+
+    def get_viewer_cfg(self) -> ViewerCfg:
+        return get_viewer_cfg_look_at_object(
+            lookat_object=self.lift_object,
+            offset=np.array([-1.5, -1.5, 1.5]),
+        )
+
+
+@configclass
+class LiftObjectEventsCfg:
+    """Configuration for Lift Object."""
+
+    reset_lift_object_pose: EventTermCfg = MISSING
+
+    def __init__(self, lift_object: Asset):
+        self.reset_lift_object_pose = EventTermCfg(
+            func=mdp_isaac_lab.reset_root_state_uniform,
+            mode="reset",
+            params={
+                "pose_range": {
+                    "x": (-0.1, 0.1),
+                    "y": (-0.25, 0.25),
+                    "z": (0.0, 0.0),
+                },
+                "velocity_range": {},
+                "asset_cfg": SceneEntityCfg(lift_object.name),
+            },
+        )
+
+
+@configclass
+class LiftObjectTerminationsCfg:
+    """Termination terms for the Lift Object task."""
+
+    time_out: TerminationTermCfg = TerminationTermCfg(func=mdp_isaac_lab.time_out)
+    object_dropped: TerminationTermCfg = MISSING
+
+
+class LiftObjectTaskRL(LiftObjectTask):
+    def __init__(
+        self,
+        lift_object: Asset,
+        background_scene: Asset,
+        embodiment: EmbodimentBase,
+        minimum_height_to_lift: float = 0.04,
+        episode_length_s: float = 5.0,
+    ):
+        super().__init__(
+            lift_object=lift_object,
+            background_scene=background_scene,
+            minimum_height_to_lift=minimum_height_to_lift,
+            episode_length_s=episode_length_s,
+        )
+        self.embodiment = embodiment
+
+    def get_observation_cfg(self):
+        return LiftObjectObservationsCfg(lift_object=self.lift_object)
+
+    def get_rewards_cfg(self):
+        return LiftObjectRewardCfg(
+            lift_object=self.lift_object,
+            minimum_height_to_lift=self.minimum_height_to_lift,
+            robot_name=self.embodiment.get_embodiment_name_in_scene(),
+            ee_frame_name=self.embodiment.get_ee_frame_name(),
+        )
+
+    def get_commands_cfg(self):
+        return LiftObjectCommandsCfg(
+            asset_name=self.embodiment.get_embodiment_name_in_scene(),
+            body_name=self.embodiment.get_command_body_name(),
+            lift_object=self.lift_object,
+        )
```

Comment:
What do you think about following our "maximum modifyability" policy that we pursue elsewhere, where we build configs in the constructor, such that a user can modify after the fact.